src/common/include/com/amazonaws/kinesis/video/common/CommonDefs.h

```diff
@@ -145,8 +145,8 @@ typedef unsigned char UINT8;
 typedef char INT8;
 typedef unsigned short UINT16;
 typedef short INT16;
-typedef unsigned long UINT32;
-typedef long INT32;
+typedef unsigned int UINT32;
+typedef int INT32;

```

Comment:
The C standard only guarantees that an int be at least 16 bits so this can fail on some platforms.

Reply:
Thanks for calling this out, I added a conditional to check for the size of `UINT_MAX` which comes from `limits.h` which, from my findings, should be available on all platforms.

Reply:
Ideally, if needed needed, INT32 should define to `int32_t` from `inttypes.h` guaranteeing it to be always 32 bits and likewise for other macros which have exact sizes.